pkg/sql/create_index.go

```diff
@@ -243,6 +247,17 @@ func MakeIndexDescriptor(
 	if err := indexDesc.FillColumns(n.Columns); err != nil {
 		return nil, err
 	}
+
+	paramObserver := indexStorageParamObserver{indexDesc: &indexDesc}
```

Comment:
nit i'd just inline the creation of this observer

Reply:
BUT MY HEAP
ok